src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ExclusionZoneService.kt

```diff
@@ -0,0 +1,182 @@
+      fileType = file.contentType,
+      fileSize = file.size.toInt(),
+      description = description,
+      thumbnailImage = thumbnailImage,
+      uploadDetailId = savedDetail.id,
+    )
+
+    val updatedAdditionalCondition = additionalCondition.copy(additionalConditionUploadSummary = listOf(uploadSummary))
+
+    additionalConditionRepository.saveAndFlush(updatedAdditionalCondition)
+  }
+
+  @Transactional
+  fun removeExclusionZoneFile(licenceId: Long, conditionId: Long) {
+    licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val additionalCondition = additionalConditionRepository
+      .findById(conditionId)
+      .orElseThrow { EntityNotFoundException("$conditionId") }
+
+    // Remove uploadDetail rows manually - it is intentionally not linked to the additionalCondition entity
+    additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }.forEach {
+      additionalConditionUploadDetailRepository.findById(it).ifPresent { detail ->
+        additionalConditionUploadDetailRepository.delete(detail)
+      }
+    }
+
+    // Remove the uploadSummary via the additionalCondition
+    val updatedAdditionalCondition = additionalCondition.copy(additionalConditionUploadSummary = emptyList())
+    additionalConditionRepository.saveAndFlush(updatedAdditionalCondition)
+  }
+
+  fun getExclusionZoneImage(licenceId: Long, conditionId: Long): ByteArray? {
+    licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val additionalCondition = additionalConditionRepository
+      .findById(conditionId)
+      .orElseThrow { EntityNotFoundException("$conditionId") }
+
+    val uploadIds = additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }
+    if (uploadIds.isEmpty()) {
+      throw EntityNotFoundException("$conditionId")
+    }
+
+    val upload = additionalConditionUploadDetailRepository
+      .findById(uploadIds.first())
+      .orElseThrow { EntityNotFoundException("$conditionId") }
+
+    return upload.fullSizeImage
+  }
+
+  fun extractFullSizeImageJpeg(fileStream: InputStream): ByteArray? {
+    var pdfDoc: PDDocument? = null
+    try {
+      pdfDoc = PDDocument.load(fileStream)
+      val renderer = PDFRenderer(pdfDoc)
+      val firstImage = renderer.renderImage(0)
+      val baos = ByteArrayOutputStream()
+      ImageIO.write(firstImage, "jpg", baos)
+      return baos.toByteArray()
+    } catch (e: Exception) {
+      log.error("Extracting full size image - error ${e.message}")
+    } finally {
+      pdfDoc?.close()
+      fileStream.close()
+    }
+    return ByteArray(0)
+  }
+
+  fun extractDescription(fileStream: InputStream): String? {
+    var pdfDoc: PDDocument? = null
+    try {
+      pdfDoc = PDDocument.load(fileStream)
+      val stripper = PDFTextStripper()
+      stripper.sortByPosition = true
+      stripper.startPage = 2
```

Comment:
Can you be sure that the description will be on page 2? Never on page 1 even?

Reply:
Yes!  I don't like it, as its very specific to this file created by MapMaker, which always puts the large area map on Page 1 and the descriptive text on Page 2.